circle.yml

```diff
@@ -1,35 +1,74 @@
-machine:
-  python:
-    version: 3.5.3
-  services:
-    - redis
-  environment:
-    # Don't add sensitive information (e.g. passwords) here but in https://circleci.com/gh/open-craft/opencraft/edit#env-vars
-    DEBUG: 'true'
-    DEFAULT_FORK: 'open-craft/edx-platform'
-    INSTANCE_EPHEMERAL_DATABASES: 'false'
-    LOAD_BALANCER_FRAGMENT_NAME_PREFIX: 'integration-'
-    TEST_RUNNER: 'opencraft.tests.utils.CircleCIParallelTestRunner'
-dependencies:
-  pre:
-    - pip install --upgrade pip
-    - pip install --upgrade virtualenv
-    # Update python to latest 2.6.* release from the deadsnakes PPA.
-    # Ubuntu 14.04 which we currently use on CircleCI comes with 2.7.6 preinstalled,
-    # which does not support SNI. When CircleCI starts providing Ubuntu 16.04 images,
-    # this won't be needed anymore.
-    - sudo add-apt-repository ppa:fkrull/deadsnakes-python2.7 -y
-    - sudo apt-get update; sudo apt-get install python2.7 python2.7-dev
-  override:
-    - pip install -r requirements.txt
-database:
-  override:
-    - make create_db
-test:
-  override:
-    - bin/run-circleci-tests:
-        timeout: 2800
-        parallel: true
-  post:
-    - make test.integration_cleanup:
-        timeout: 1200
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/python:3.5.4-browsers
+        environment:
+          DEBUG: 'true'
+          DEFAULT_FORK: 'open-craft/edx-platform'
+          INSTANCE_EPHEMERAL_DATABASES: 'false'
+          LOAD_BALANCER_FRAGMENT_NAME_PREFIX: 'integration-'
+          TEST_RUNNER: 'opencraft.tests.utils.CircleCIParallelTestRunner'
+      - image: redis
+      - image: mongo
+      - image: mysql
+        environment:
+          MYSQL_ALLOW_EMPTY_PASSWORD: True
+      - image: circleci/postgres:10.1-alpine
+        environment:
+          PG_HOST: 127.0.0.1
+          POSTGRES_USER: circleci
+          POSTGRES_DB: circle_test
+    parallelism: 2
+    steps:
+      - checkout
+      - restore_cache:
+          key: deps1-{{ .Branch }}-{{ checksum "requirements.txt" }}
+      - run:
+          name: Install Dockerize
+          command: wget https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && sudo tar -C /usr/local/bin -xzvf dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && rm dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz
```

Comment:
This is only used for waiting for the ports to be open, right?  It may be easier to use `netcat` instead, which I believe is already installed even in minimal Ubuntu images.  (I'm fine with what you have – just suggesting an alternative solution you can consider.)

Reply:
Agree, much better 👍